### ISSUE TITLE
GraphiteMeterRegistry namingConvention to be independent of graphiteTagsEnabled

### ISSUE THREAD

**Issue by nikobearrr** [micrometer-metrics/micrometer#3090]
**Please describe the feature request.**

It looks like I am not able to configure which `HierarchicalNameMapper` to use since it is bound to `config.graphiteTagsEnabled()`.
In the `GraphiteMeterRegistry` there are several constructors which take `HierarchicalNameMapper nameMapper` as argument, but are overwritten in the constructor call.

https://github.com/micrometer-metrics/micrometer/blob/v1.8.3/implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteMeterRegistry.java#L51

This line makes it irrelevant (to my knowledge) which kind of naming convention I would like to use. I would like to achieve, what I used to have with an older (1.5.x) micrometer version.

**Rationale**

I do not use tags on graphite and I do not want to use them.
I would like to have my metrics for springboot in a following manner:

`~/graphite/data/service/executor/completed/name/applicationTaskExecutor/m1_rate.wsp`

## 1. Metrics with tags

Currently, if I use the following code:

```
@Bean
public GraphiteMeterRegistry graphiteMeterRegistry(final GraphiteConfig config, final Clock clock) {
    return new GraphiteMeterRegistry(config, clock, new GraphiteHierarchicalNameMapper("service", "extraPrefix"));
}
```

The metrics I receive on graphite look like this:

`~/graphite/data/myservice/executor/completed/name/applicationTaskExecutor;metricattribute=m1_rate.wsp`

## 2. Metrics without tags

The default of `GraphiteConfig` is to set `graphiteTagsEnabled()` to `true`. I can set it to `false` using `management.metrics.export.graphite.graphite-tags-enabled=false`

The metrics look like this:

`~/graphite/data/myservice/executorCompleted/name/applicationTaskExecutor/m1_rate.wsp`

## 3. Using metrics without tags and with custom name mapper

The only way I found to achieve what I am going for, is if I write this:

```
@Bean
public GraphiteMeterRegistry graphiteMeterRegistry(final GraphiteConfig config, final Clock clock) {
    MetricRegistry metricRegistry = new MetricRegistry();
    return new GraphiteMeterRegistry(config, clock, new GraphiteHierarchicalNameMapper("service", "extraPrefix"), metricRegistry, defaultGraphiteReporter(config, clock, metricRegistry));
}

private static GraphiteReporter defaultGraphiteReporter(GraphiteConfig config, Clock clock, MetricRegistry metricRegistry) {
    return GraphiteReporter.forRegistry(metricRegistry)
            .withClock(new DropwizardClock(clock))
            .convertRatesTo(config.rateUnits())
            .convertDurationsTo(config.durationUnits())
            .addMetricAttributesAsTags(false)
            .build(getGraphiteSender(config));
}

private static GraphiteSender getGraphiteSender(GraphiteConfig config) {
    switch (config.protocol()) {
        case PLAINTEXT:
            return new Graphite(config.host(), config.port());
        case UDP:
            return new GraphiteUDP(config.host(), config.port());
        case PICKLED:
        default:
            return new PickledGraphite(config.host(), config.port());
    }
}
```

This is a copy-paste from the original GraphiteMeterRegistry https://github.com/micrometer-metrics/micrometer/blob/v1.8.3/implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteMeterRegistry.java#L57

This produces the following metrics:

`~/graphite/data/myservice/executor/completed/name/applicationTaskExecutor/m1_rate.wsp`

**Additional context**

I am running springboot 2.6.4

What I am writing now seems to have been discussed here: https://github.com/micrometer-metrics/micrometer/issues/2069#issuecomment-630620342

The issue with the metrics from 1) is that they are appended with a tag (tag enabled is true)
The issue with the metrics from 2) is that they do not obey the naming convention I want to use and overwrite it inside the constructor. 

Bottom line is I would like the namingConvention to not be dependent on the `config.graphiteTagsEnabled()` https://github.com/micrometer-metrics/micrometer/blob/v1.8.3/implementations/micrometer-registry-graphite/src/main/java/io/micrometer/graphite/GraphiteMeterRegistry.java#L51

**Comment by marcingrzejszczak** [comment]
Do I understand correctly that you would like to be able to provide your own `NamingConvention` and currently it's always overridden due to what's happening in the constructor?

You can always call or am I missing sth?

```java
new GraphiteMeterRegistry(...).config().namingConvention(your new convention here)
```
